### PR TITLE
`Development`: Add git commit to navbar version in dev mode

### PR DESCRIPTION
**Pull request by TheZoker**
#### General

- [ ] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] This is a small issue that I tested locally
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.ase.in.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).

### Description

This adds a git commit id into the header, when Artemis is run in development mode. 

image

Not displayed in the prod mode:
image

### Steps for Testing

1. Start Artemis in dev mode
2. Have a look at the version in the header

#### Code Review
- [ ] Review 1
- [ ] Review 2
#### Manual Tests
- [ ] Test 1
- [ ] Test 2

### Test Coverage

| Class/File | Branch | Line |
|------------|-------:|-----:|

### CODE REVIEW

**Review by ge65cer — COMMENTED** [review]
I don't think you need a new service class for this. You can probably integrate it into the existing `ProfileService` which is already shared between components.
Or is there a specific reason you implemented it this way?

**Review by bassner — CHANGES_REQUESTED** [review]
I think the same as @ge65cer. We should not introduce a new service for this. This will duplicate the REST call to the management info endpoint (one from the profile service, one from the new one). Just add the commit and branch as new properties in the profile service

**Review by Hialus — CHANGES_REQUESTED** [review]
In addition to what Moritz and Patrick said:
I personally don't think it makes much sense just to include the commit hash here. I think showing the branch from which this release was created as well would make more sense. In order to not clutter it you might show one of the values only when hovering over the version.
It's just an idea though, as I think this might be useful to know, as currently the only way to find out what branch is currently deployed is to check it on Bamboo.

**Review by ge65cer — CHANGES_REQUESTED** [review]
See my comment above.

**Comment by TheZoker** [comment]
I removed the additional service and moved to the already existing profile service. Thanks for the hints!

Since [this PR](https://github.com/ls1intum/Artemis/pull/5580), you can see which branch was used by just visiting the `/managment/info` page and it will tell you a lot of information. I think having the git commit in the header when in dev (as suggested by Stefan) is a good idea and if you want to have more information about git, you can look it up in the mentioned endpoint. And [this PR](https://github.com/ls1intum/Artemis/pull/5607) also added the branch name to the startup log.

**Review by jpbernius — APPROVED** [review]

**Review by ge65cer — COMMENTED** [review]
Thanks for the refactor, a lot cleaner now! I tested on TS2 and the endpoint returns the branch name correctly.

It does not display the commit id in the navbar on our test servers, though. (I suppose they run in production mode.) In my opinion, that would be very useful. At best, I could hover over the version and a tooltip would show me commit id and branch name.

**Review by Hialus — DISMISSED** [review]
Looks good to me. Nice feature.

**Comment by krusche** [comment]
I would display the branch name immediately on dev and test systems.
The commit id should be shown in a tooltip on such systems.

On production systems, we should **not** show the commit as tooltip!

Instead, I suggest to show the branch and commit id in the bottom of the about page.

**Comment by krusche** [comment]
I would display it directly right to the version number

**Comment by TheZoker** [comment]
The branch name in the navbar:
image

On mobile devices and on production system, the branch name will be hidden.

And the about page:
image

**Review by stefanwaldhauser — APPROVED** [review]
Great job! Works perfectly. Tested it locally in dev mode and on the testserver in production mode!

File: src/main/webapp/app/core/about-us/about-us.component.html [hunk 1]
```diff
@@ -56,4 +56,8 @@ <h3 class="font-weight-bold">{{ 'artemisApp.aboutUsOverview.contributors' | arte
             </div>
         </div>
     </div>
+    <div class="mt-5 ms-2 me-2">
+        <h3 class="font-weight-bold">Git Information</h3>
+        <div>Git Branch: {{ gitBranchName }}<br />Git Commit ID: {{ gitCommitId }}</div>
```
Comment (ge65cer) [inline src/main/webapp/app/core/about-us/about-us.component.html:61]: Even though the translations might be identical here, please move all text to the translation files.

**Review by ge65cer — CHANGES_REQUESTED** [review]
Looks good! Translations & client tests for the about us page need to be adjusted.
As the space taken by the version is now significantly longer in the navbar, this might mess up the responsive layout there and the breakpoints may need to be adjusted. CC @bassner

File: src/main/webapp/app/core/about-us/about-us.component.html (hunk 1 above)
Comment (TheZoker) [reply src/main/webapp/app/core/about-us/about-us.component.html]: Done :)

**Review by ge65cer — APPROVED** [review]
Looks good from my side now. But as I said please wait for the okay from @bassner regarding the navbar.

**Comment by bassner** [comment]
Well, yes, this is messing up the breakpoints; if you're adding stuff to the navbar that consumes width, the calculation for the breakpoints needs to be adjusted [here](https://github.com/ls1intum/Artemis/blob/be91820c4fd7d21ed20f4498ec4af4fe87ce2d8f/src/main/webapp/app/shared/layouts/navbar/navbar.component.ts#L146L173).

> At best, I could hover over the version and a tooltip would show me commit id and branch name.

This would probably be the simplest solution for that problem

https://user-images.githubusercontent.com/23171488/192281821-38cf5bfd-f56d-494f-a0bc-e239492dd69d.mov

**Comment by TheZoker** [comment]
Thanks for the hint!
The hover style was actually the first approach. 

@krusche
Should I revert it back to hover or do you want to display the information right away?

**Comment by krusche** [comment]
> Thanks for the hint! The hover style was actually the first approach.
> 
> @krusche Should I revert it back to hover or do you want to display the information right away?

If it is too complicated to show it in the navbar directly, then I suggest to add it to the bottom of the page (branch + git commit) above or below the following footer
image

Of course, this should only be shown for dev and test environments

If you want to show it additionally when hovering over the version, this would be fine for me, but I'd like to see it quickly on dev and test environments without the need to move the mouse ;-)

**Comment by krusche** [comment]
I would change `Git Commit ID: a8482b6` to simply `Git commit: a8482b6`

**Pull request closed without merging**